resources/uk/gov/hmcts/pipeline/yarn/yarn-audit-with-suppressions.sh

```diff
@@ -129,13 +136,26 @@ today=$(date +"%s")
 exclude_until="1708502400"
 
 if [ "$YARN_VERSION" != "4" ]; then
+  IS_YARN4=0
   print_upgrade_yarn4
+else
+  IS_YARN4=1
+fi
+
+yarn_audit_command="yarn npm audit --recursive --environment production --json"
+if [ "$IS_YARN4" -eq 0 ]; then
+  echo "You have an older version of Yarn that has issues with --recursive flag, so we will run '--all' instead"
+  yarn_audit_command="yarn npm audit --all --environment production --json"
```

Comment:
Could we not just move this to the else block above? Rather than having two `if` statements effectively doing the same thing?